### ISSUE TITLE
NullPointerException when executing Paparazzi tests via Android Studio

### ISSUE THREAD

**Issue by alteral** [cashapp/paparazzi#175]
**Environment:**
- Paparazzi: 0.6.0
- Android API: less than 30 (e.g.: 29)
- OpenJDK: 11.0.8
- Mac OS: 10.15.4
- Android Studio: 4.0.2

**Steps:**
- Clone [paparazzi repository](https://github.com/cashapp/paparazzi)
- Open paparazzi project via Android Studio
- Open any sample test class file (e.g.: [LaunchViewTest.kt](https://github.com/cashapp/paparazzi/blob/master/sample/src/test/java/app/cash/paparazzi/sample/LaunchViewTest.kt))
- Run any test individually, tapping on `run` button (e.g.: `pixel3`)

**Actual result:**
- NullPointerException:
```
java.lang.NullPointerException
	at java.base/java.io.File.<init>(File.java:278)
	at app.cash.paparazzi.EnvironmentKt.detectEnvironment(Environment.kt:36)
	at app.cash.paparazzi.Paparazzi.<init>(Paparazzi.kt:57)
	at app.cash.paparazzi.sample.LaunchViewTest.<init>(LaunchViewTest.kt:29)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:250)
	at org.junit.runners.BlockJUnit4ClassRunner.createTest(BlockJUnit4ClassRunner.java:260)
	at org.junit.runners.BlockJUnit4ClassRunner$2.runReflectiveCall(BlockJUnit4ClassRunner.java:309)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.BlockJUnit4ClassRunner.methodBlock(BlockJUnit4ClassRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
```

**Comment by jrodbx** [comment]
Spent some time investigating this.  Currently, running a test via the run button or via a run configuration in Android Studio will run up to the compile tasks using Gradle, then run the tests themselves via an internal JUnit Runner, instead of using the Gradle test task. The Paparazzi Gradle plugin sets JVM system properties for that test task and so it looks like this is currently broken via the IDE.

While one approach could be to rewrite the plugin is pass these properties differently, I'm encouraged that [this recent survey](https://forms.gle/qwpcscH7JtgAtN2G9) by the Android Studio Tools team will result in the IDE following the same code execution path as CLI test runs in an upcoming version.

"Android Studio JUnit Testing Survey
The Android Studio team is planning to migrate away from Android JUnit as a test type for unit tests you run from the IDE, and will run these tests via Gradle instead. We’re doing this because the Android JUnit test runner has become increasingly inconsistent in the way it executes tests, resulting in lost productivity and frustrations for users. Additionally, switching to running tests via Gradle would also improve consistency between Android Studio and tests you run on your CI server. A result of this migration is that any manually created Android JUnit test configurations (i.e. the configurations you create from Run > Edit Configurations) will no longer be recognized by Android Studio. While we think it’s in the best interest of our users to migrate to running their tests via Gradle as soon as possible, we want to gather feedback on how disruptive this change might be and how many users depend on the custom Android JUnit configurations for their day-to-day productivity."

**Comment by jrodbx** [comment]
This has been fixed in Android Studio Arctic Fox | 2020.3.1 Canary 2

Android Studio 4.1.1
```
Executing tasks: [:sample:generateDebugSources, :sample:compileDebugSources, :sample:createMockableJar, :sample:compileDebugUnitTestSources] in project /Users/jrod/Development/paparazzi
```

Android Studio Arctic Fox | 2020.3.1 Canary 2:
```
Executing tasks: [:sample:cleanTestDebugUnitTest, :sample:testDebugUnitTest] in project /Users/jrod/Development/paparazzi
```
